File: 0485-max-consecutive-ones/0485-max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        result=consecutive=0
        for i in nums:
            consecutive=consecutive*i+i
            result=max(consecutive,result)
        return result    
"""        count=0
        max_count=0
        for i in nums:
            if i==1:
                count+=1
            else:
                if max_count<count:
                    max_count=count
                count=0
        if max_count<count:
            max_count=count
        return max_count    """    

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int count=0;
        int maxCount=0;
        int n=nums.size();
        for(int i=0;i<n;i++){
            if(nums[i]==1){
                count++;
            }
            else{
                count=0;
            }
            maxCount=max(maxCount,count);
        }
        return maxCount;
    }
};